src/Aspire.Hosting.Azure.AIFoundry/AzureAIFoundryLocalResourceExtensions.cs

```diff
@@ -0,0 +1,259 @@
+    {
+        builder.ApplicationBuilder.Eventing.Subscribe<InitializeResourceEvent>(builder.Resource, (@event, ct)
+            => Task.Run(async () =>
+            {
+                var resource = (AzureAIFoundryResource)@event.Resource;
+                var rns = @event.Services.GetRequiredService<ResourceNotificationService>();
+                var manager = @event.Services.GetRequiredService<FoundryLocalManager>();
+
+                await rns.PublishUpdateAsync(resource, state => state with
+                {
+                    State = new ResourceStateSnapshot(KnownResourceStates.Starting, KnownResourceStateStyles.Info)
+                }).ConfigureAwait(false);
+
+                await manager.StartServiceAsync(ct).ConfigureAwait(false);
+
+                resource.ApiKey = manager.ApiKey;
+
+                if (manager.IsServiceRunning)
+                {
+                    if (resource.TryGetLastAnnotation<EndpointAnnotation>(out var endpoint))
+                    {
+                        endpoint.AllocatedEndpoint = new AllocatedEndpoint(
+                            new EndpointAnnotation(
+                                System.Net.Sockets.ProtocolType.Tcp,
+                                manager.Endpoint.Scheme,
+                                "http",
+                                "http",
+                                manager.Endpoint.Port,
+                                manager.Endpoint.Port,
+                                false,
+                                false),
+                            manager.Endpoint.ToString(),
+                            manager.Endpoint.Port);
+                    }
+
+                    await rns.PublishUpdateAsync(resource, state => state with
+                    {
+                        State = KnownResourceStates.Running,
+                        Properties = [.. state.Properties, new(CustomResourceKnownProperties.Source, "Foundry Local")]
+                    }).ConfigureAwait(false);
+                }
+                else
+                {
+                    await rns.PublishUpdateAsync(resource, state => state with
+                    {
+                        State = KnownResourceStates.FailedToStart,
+                        Properties = [.. state.Properties, new(CustomResourceKnownProperties.Source, "Foundry Local")]
+                    }).ConfigureAwait(false);
+                }
+
+            }, ct));
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Ensures that the resource stops when the application is shutting down.
+    /// </summary>
+    /// <param name="resource">The resource builder for the Foundry Local resource.</param>
+    /// <returns>The updated resource builder.</returns>
+    private static IResourceBuilder<AzureAIFoundryResource> EnsureResourceStops(this IResourceBuilder<AzureAIFoundryResource> resource)
```

Comment:
Let's remove this method entirely. I originally wrote it with the theory that you could run multiple instances of Foundry Local (since it was an executable as the entry point) but have since learnt that that isn't possible, so trying to force a stop will just cause problems.